cranelift/interpreter/src/step.rs

```diff
@@ -155,10 +181,15 @@ where
         Opcode::BrIcmp => branch_when(icmp(inst.cond_code().unwrap(), &arg(0)?, &arg(1)?)?)?,
         Opcode::Brif => branch_when(state.has_iflag(inst.cond_code().unwrap()))?,
         Opcode::Brff => branch_when(state.has_fflag(inst.fp_cond_code().unwrap()))?,
-        Opcode::BrTable => unimplemented!("BrTable"),
-        Opcode::JumpTableEntry => unimplemented!("JumpTableEntry"),
-        Opcode::JumpTableBase => unimplemented!("JumpTableBase"),
-        Opcode::IndirectJumpTableBr => unimplemented!("IndirectJumpTableBr"),
+        Opcode::BrTable => branch_indexed(arg(0)?),
+        // `jump_table_base` returns the base address of a jump table. We don't have that in the
+        // interpreter, but we only need this to index into a jump table, so we always
+        // return a base index of 0, to be used in `indirect_jump_table_br`
+        Opcode::JumpTableBase => assign(Value::int(0, ctrl_ty)?),
+        // See the comment in `JumpTableBase`. Continuing that logic, here we only return
+        // the offset of the entry + the base (which is always 0).
+        Opcode::JumpTableEntry => binary(Value::add, arg(0)?, arg(1)?)?,
```

Comment:
There are more arguments to this instruction, namely `size` and the jump table.

* Should we perform any sort of assertion on `size`?

* If the user tries to request an entry out of bounds, do we allow them to do that and then trap in `IndirectJumpTableBr `? Or do we prevent that entry from being generated here?